geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -262,30 +258,26 @@ long zcount(SortedSetScoreRangeOptions rangeOptions) {
     return maxIndex - minIndex;
   }
 
-  byte[] zincrby(Region<RedisKey, RedisData> region, RedisKey key, byte[] increment,
-      byte[] member) {
+  byte[] zincrby(Region<RedisKey, RedisData> region, RedisKey key, double incr, byte[] member) {
+    double score;
     OrderedSetEntry orderedSetEntry = members.get(member);
-    double incr = processByteArrayAsDouble(increment);
 
     if (orderedSetEntry != null) {
-      double byteScore = orderedSetEntry.getScore();
-      incr += byteScore;
+      score = orderedSetEntry.getScore() + incr;
 
-      if (Double.isNaN(incr)) {
+      if (Double.isNaN(score)) {
         throw new NumberFormatException(RedisConstants.ERROR_OPERATION_PRODUCED_NAN);
       }
+    } else {
+      score = incr;
     }
 
-    byte[] byteIncr = doubleToBytes(incr);
-    memberAdd(member, byteIncr);
-
-    AddsDeltaInfo deltaInfo = new AddsDeltaInfo(new ArrayList<>());
-    deltaInfo.add(member);
-    deltaInfo.add(byteIncr);
-
-    storeChanges(region, key, deltaInfo);
+    if (!(memberAdd(member, score) == MemberAddResult.NO_OP)) {
+      ZAddsDeltaInfo deltaInfo = new ZAddsDeltaInfo(1);
+      storeChanges(region, key, deltaInfo);
```

Comment:
The failing DUnit test can be fixed by making this
```
      ZAddsDeltaInfo deltaInfo = new ZAddsDeltaInfo(1);
      deltaInfo.add(member, score);
      storeChanges(region, key, deltaInfo);
```

Reply:
thanks